brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -110,6 +110,9 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
         req = req.withHeaders(newHeaders);
         ctx.updateRequest(req);
 
+        // Make the span the "current span" when the ctx is pushed
```

Comment:
The span is already current by `RequestContextCurrentTraceContext` so this is only trying to run the decorators - we added that class mostly to avoid this hook :P 

I think you can expose this method to the decorators

https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/propagation/CurrentTraceContext.java#L106

and call it instead.